jbpm-flow/src/main/java/org/jbpm/process/instance/ProcessVariables.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.process.instance;
+
+import java.beans.BeanInfo;
+import java.beans.FeatureDescriptor;
+import java.beans.IntrospectionException;
+import java.beans.Introspector;
+import java.beans.PropertyDescriptor;
+import java.io.Serializable;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.jbpm.process.core.context.variable.LambdaValueReference;
+import org.jbpm.process.core.context.variable.SimpleValueReference;
+import org.jbpm.process.core.context.variable.ValueReference;
+import org.kie.api.runtime.rule.RuleUnit;
+
+public abstract class ProcessVariables implements Serializable {
+
+    public static <T extends RuleUnit> ProcessVariables.Typed<T> typed(T value) {
```

Comment:
Same question here, what rule unit is used for? In general maybe we should abstract it bit to have a unit instead rule unit as it might make more sense to refer to unit or process unit in jbpm

Reply:
removed

Reply:
good catch